openmp/libomptarget/plugins-nextgen/amdgpu/src/rtl.cpp

```diff
@@ -2728,6 +2749,30 @@ struct AMDGPUDeviceTy : public GenericDeviceTy, AMDGenericDeviceTy {
     return Err;
   }
 
+  /// Detect if current architecture is an APU.
+  Error checkIfAPU() {
+    std::string StrGfxName(ComputeUnitKind);
```

Comment:
Shouldn't really need std::string for a small set of constant strings?

Reply:
Should I use
const std::string StrGfxName(ComputeUnitKind);
?
Or how do you propose I change the code?

Reply:
StringRef?